### PR TITLE
Feature/new notes router+model

### PR DESCRIPTION
**Pull request by thecuriousones**
# Description
Loon Video: 
This pull request contains the model and router files for the client notes system.

Fixes #

- What work was done?
      - Model and router files were created
- Why was this work done?
      - Frontend has a way to access the backend's database containing mock client notes for the notes system
- What feature / user story is it for?
      -  A system containing all of the case managers' notes for different clients (households)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Change Status

- [ ] Complete, tested, ready to review and merge
- [x] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

## Has This Been Tested

- [x] Yes
- [ ] No
- [ ] Not necessary

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

File: api/app.js [hunk 1]
```diff
@@ -52,7 +52,7 @@ app.use(cookieParser());
 app.use('/', indexRouter);
 app.use(['/profile', '/profiles'], profileRouter);
 app.use('/data', dsRouter);
-app.use('/clients', clientRouter);
+app.use('/api/clients', clientRouter);
```
Comment (stepbystepdayo) [inline api/app.js:55]: looks good😊

File: api/client/clientModel.js [hunk 2]
```diff
@@ -24,10 +25,54 @@ const update = (id, changes) => {
   return db('clients').where({ id }).update(changes, '*');
 };
 
+// FOR THE CLIENT NOTES ROUTES
+function get(client_id) {
+  console.log('model working?');
```
Comment (stepbystepdayo) [inline api/client/clientModel.js:26]: this code looks great! should we remove console.logs?

File: api/client/clientRouter.js [hunk 3]
```diff
@@ -76,4 +76,196 @@ router.put(':id', (req, res) => {
     });
 });
 
+// CLIENT NOTES ROUTERS
+
+/**
+ * @swagger
```
Comment (stepbystepdayo) [inline api/client/clientRouter.js:82]: great job on Swagger! I will make own too soon.

File: api/client/clientModel.js [hunk 4]
```diff
@@ -24,10 +25,54 @@ const update = (id, changes) => {
...
+function insert(note) {
+  return db('client_notes')
+    .insert(note)
+    .then((id) => {
+      return getById(id[0]);
+    });
+}
+
+async function notesUpdate(clientID, noteID, changes) {
+  console.log(`the client ID here, ${clientID}, is not currently being used.`);
```
Comment (stepbystepdayo) [inline api/client/clientModel.js:43]: maybe remove console.log?

File: api/client/clientRouter.js [hunk 5]
```diff
@@ -76,4 +76,196 @@ router.put(':id', (req, res) => {
...
+ *                  message: 'Needs a few more forms filled out before approval'
+ *                  created_by: 'Jamie Kai'
+ *                  created_at: 03/04/2222 @10:03
+ *      500:
+ *        message: 'Error retrieving client notes'
+ */
+
+router.get('/:id/notes', (req, res) => {
+  const { id } = req.params;
+  console.log('working notes?');
```
Comment (stepbystepdayo) [inline api/client/clientRouter.js:110]: maybe remove console.log?

File: api/client/clientRouter.js [hunk 6]
```diff
@@ -76,4 +76,196 @@ router.put(':id', (req, res) => {
...
+ *                  created_at: 02/22/2222 @13:55
+ *      500:
+ *        message: 'Error retrieving client note'
+ */
+
+router.get('/:id/notes/:id', (req, res) => {
+  const { id } = req.params;
+  Clients.getById(id)
+    .then((note) => {
+      console.log(note);
```
Comment (stepbystepdayo) [inline api/client/clientRouter.js:149]: maybe remove console.log?

File: api/client/clientRouter.js [hunk 7]
```diff
@@ -76,4 +76,196 @@ router.put(':id', (req, res) => {
...
+ *                  created_by: 'Jon Adams'
+ *                  created_at: 05/22/2222 @18:35
+ *      500:
+ *        message: 'Error adding client note'
+ */
+
+router.post('/:id/notes', (req, res) => {
+  Clients.insert(req.body)
+    .then((note) => {
+      console.log('New note added:', note);
```
Comment (stepbystepdayo) [inline api/client/clientRouter.js:182]: maybe remove console.log?

**Review by stepbystepdayo — COMMENTED** [review]
I think looks fine but I will test it locally. after checking it I will approve it. great job Tamara❤️‍🔥

File: api/client/clientRouter.js [hunk 8]
```diff
@@ -76,4 +76,196 @@ router.put(':id', (req, res) => {
...
+ *                  client_id: '7'
+ *                  source_view: 'home dashboard'
+ *                  message: 'Family is waiting for a response from the courthouse'
+ *                  created_by: 'Tamara Leonard'
+ *                  created_at: 02/22/2222 @13:55
+ *      500:
+ *        message: 'Error retrieving client note'
+ */
+
+router.get('/:id/notes/:id', (req, res) => {
```
Comment (stepbystepdayo) [inline api/client/clientRouter.js:145]: I think we cannot use ":id" twice here. I think we need to use `:clientID/notes/:noteID`?

**Comment by gsauce7** [comment]
Looks good, but we definitely want to remove all "to-dos" and console.logs in the final version of the code

**Comment by derekjpeters** [comment]
[New Notes -  Router + Model](https://trello.com/c/f7ICD4pu/66-new-notes-router-model)

**Pull request merged**